src/main/java/org/grouplens/grapht/util/LogContext.java

```diff
@@ -0,0 +1,46 @@
+package org.grouplens.grapht.util;
+/**
+ * Created by Imran Kazi on 2/9/2015.
+ */
+import org.slf4j.MDC;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Utility class to manage log data with {@link MDC}.  This class allows MDC parameters
+ * to be set, and popped back off when the context is finished.
+ */
+public class LogContext {
+    @SuppressWarnings("rawtypes")
+
+
+    private final Map memory = MDC.getCopyOfContextMap();
+
+    //--- Make the default constructor non-public.
+    private  LogContext() {}
+
+    //--- instantiating LogContext class with create method.
```

Comment:
We should replace this with a JavaDoc comment that indicates that this method creates a new log context, capturing the MDC's current data to be restored when `finish()` is called.